homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -747,6 +777,10 @@ def __init__(self, name, device, model, unique_id):
             self._device_features = FEATURE_FLAGS_AIRPURIFIER_2S
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRPURIFIER_2S
             self._speed_list = OPERATION_MODES_AIRPURIFIER_2S
+        elif self._model == MODEL_AIRPURIFIER_2H:
+            self._device_features = FEATURE_FLAGS_AIRPURIFIER_2H
+            self._available_attributes = AVAILABLE_ATTRIBUTES_AIRPURIFIER_2H
+            self._speed_list = OPERATION_MODES_AIRPURIFIER_2H
```

Comment:
We only allow standard fan speed constants in the speed list and as speed attribute.

https://github.com/home-assistant/core/blob/51da605b9f35d29e9b00fbf9a4eae0aca9123c3c/homeassistant/components/fan/__init__.py#L40-L43

Reply:
As I said on the other comment, this is a common setup for all fan devices in the xiaomi_miio component. Furthermore, the constants you provide cannot be translated to speeds the purifier devices understand. For example: the 2H has a "favorite" speed that is a constant speed previously defined by the user.

Reply:
This platform doesn't currently follow our architecture design. We won't allow it to further break our architecture by extending it incorrectly.

Reply:
I understand. However, I am not experienced enough to make such changes, nor do I have all the miio devices to test them. Feel free to close this PR.

Reply:
Ok, I'll close.